onnxruntime/core/framework/session_options.h

```diff
@@ -129,6 +129,12 @@ struct SessionOptions {
 
   // custom function callback to join a thread
   OrtCustomJoinThreadFn custom_join_thread_fn = nullptr;
+
+  // if true, all inconsistencies encountered during shape and type inference
+  // will result in failures. If false, in some cases warnings will be logged
+  // but processing will continue.
+  // may be useful to expose bugs in models.
+  bool strict_shape_type_inference = false;
 };
```

Comment:
This feels way to visible for something that will be rarely used. Could it use ConfigOptions instead? That would also have the benefit of providing a state of 'not specified'

Reply:
Done.